tools/calyx-writer/lib.rs

```diff
@@ -353,15 +362,22 @@ impl CalyxWriter for Cell {
 /// A guard for an [`Assignment`].
 #[derive(PartialEq, Eq, Debug)]
 pub enum Guard {
-    None,
-    Constant(u64),
+    True,
     Port(Port),
     And(Box<Guard>, Box<Guard>),
 }
 
 impl CalyxWriter for Guard {
-    fn write(&self, _f: &mut IndentFormatter<'_, '_>) -> fmt::Result {
-        // TODO
+    fn write(&self, f: &mut IndentFormatter<'_, '_>) -> fmt::Result {
+        match self {
+            Guard::True => {}
+            Guard::Port(port) => port.write(f)?,
+            Guard::And(lhs, rhs) => {
+                lhs.write(f)?;
+                write!(f, " & ")?;
+                rhs.write(f)?;
```

Comment:
Just a note that if you add the other operators you'll need to deal with parentheses. A safe option is always to over-parenthesize. Or you can take a look at the magic the IR printer does (which defines an ordering on guards)

https://github.com/calyxir/calyx/blob/6c2d82cd5224f32321dd0bd9b7a867e7f0e9c804/calyx-ir/src/printer.rs#L724